tutorial/README.md

```diff
@@ -0,0 +1,413 @@
+
+`salesAGGDF2` DataFrame is created by joining `eventsDF` and `salesAGGDF2` where we register `salesAGGDFWithSchema` with a custom schema and register it as a temporary table `redshift_sales_agg` before saving it to Redshift with name  `redshift_sales_agg`
+  
+Next consider Listing 7
+
+```scala
+/*Provide a custom schema*/
+val schema = StructType(List(StructField("eventid",IntegerType,false),
+                                StructField("eventname",StringType,false),
+                                StructField("qtysold",LongType,false),
+                                StructField("revenue",DecimalType(38,2),false)
+                                ))
+val salesAGGDFWithSchema = sqlContext.createDataFrame(salesAGGDF2.rdd, schema)
+salesAGGDFWithSchema.registerTempTable("redshift_sales_agg")
+
+sqlContext.sql("select * from redshift_sales_agg")
+             .write.format("com.databricks.spark.redshift")
+             .option("url", jdbcURL) 
+             .option("tempdir", tempS3Dir) 
+             .option("dbtable", "redshift_sales_agg") 
+             .mode(SaveMode.Overwrite)
+             .save()
+```
+Listing 7: Providing Custom Schema
+
+Note how we have specified every column as non-nullable in the custom schema.
+
+In this section we illustrated how Data Sources API provides a unified API to work with multiple datasources. This feature is enabled by the generic `DataFrame` interface. We also looked at how a `DataFrame` can be saved to Redshift with a user-defined schema. None of this is unique to spark-redshift. It is provided by the DataSources API for use with all compatible packages. This is the feature that makes the DataSource API such a powerful tool for ETL usecases.
+
+## Under the hood - Putting it all together##
+
+As we discussed earlier Spark SQL will introspect for a class called `DefaultSource` in the Datasources API package `com.databricks.spark.redshift`. The `DefaultSource` class implements the trait `RelationProvider` class which is provides the default load functionality for the library. It consumes the parameters provided by the user and converts it to an instance of `BaseRelation` which is implemented by the class `com.databricks.spark.redshift.RedshiftRelation`. 
+
+This `com.databricks.spark.redshift.RedshiftRelation` class is responsible for providing an `RDD` of `org.apache.spark.sql.Row` which backs the `org.apache.spark.sql.DataFrame` instance. This represents the underlying implementation for the Load functionality for the spark-redshift package where the schema is inferred from the underlying Redshift table. The functionality for performing load where the user provides the schema is supported by the trait `org.apache.spark.sql.sources.SchemaRelationProvider` and is implemented in the class `RedshiftRelation` . 
```

Comment:
Why is "Load" capitalized in this first sentence?

> "The functionality for performing load where the user provides the schema" 

This is a kind of wordy phrase and is also passive-voice. I'd like to re-work this sentence to avoid this.